src/backend/tty.rs

```diff
@@ -1236,6 +1236,67 @@ impl Tty {
         }
     }
 
+    pub fn get_gamma_length(&self, output: &Output) -> Option<u32> {
+        let node = output.user_data().get::<TtyOutputState>().unwrap().node;
+        let devices = self.devices.get(&node);
+        if let Some(device) = devices {
+            for (_connector, crtc) in device.drm_scanner.crtcs() {
+                let crt_info = device.drm.get_crtc(crtc).unwrap();
+                return Some(crt_info.gamma_length());
+            }
+        }
+        None
```

Comment:
This will get gamma length of the first enabled output, whatever that output is. Luckily, there's `crtc` in `TtyOutputState`.

```suggestion
        let tty_state = output.user_data().get::<TtyOutputState>().unwrap();
        let device = self.devices.get(&tty_state.node).ok()?;
        let crtc_info = device.drm.get_crtc(tty_state.crtc.clone()).ok()?;
        return Some(crtc_info.gamma_length());
```